Core.Collectors/IO/AdlsBulkRecordWriter.cs

```diff
@@ -173,12 +179,17 @@ protected override async Task NotifyCurrentOutputAsync()
             }
         }
 
+        protected override string GetOutputPathPrefix(DateTime dateTimeUtc)
+        {
+            return $"{base.GetOutputPathPrefix(dateTimeUtc)}_{this.uploadIndex}";
+        }
+
         private string BulkUploadToAdlsConfig(string fileName, AdlsConfig adlsConfig)
         {
             Stopwatch uploadTimer = Stopwatch.StartNew();
             string adlsDirectory = $"{adlsConfig.AdlsRoot}/{adlsConfig.Version}";
 
-            TransferStatus status = adlsConfig.AdlsClient.BulkUpload(this.localRoot, adlsDirectory);
+            TransferStatus status = adlsConfig.AdlsClient.BulkUpload(this.localRoot, adlsDirectory, shouldOverwrite: IfExists.Fail);
```

Comment:
Do we have any exception/logging for this? I know that now this is unlikely to happen but still :)

Reply:
Yes, it comes as status code as below:
```
if (status.EntriesFailed.Count != 0)
```

Reply:
Yes, it comes as status code as below:
```
if (status.EntriesFailed.Count != 0)
```